services/worker/src/worker/job_runners/split/descriptive_statistics.py

```diff
@@ -498,7 +502,11 @@ def _compute_statistics(
         logging.info(f"Compute statistics for string column {column_name} with polars. ")
         nan_count, nan_proportion = nan_count_proportion(data, column_name, n_samples)
         n_unique = data[column_name].n_unique()
-        if n_unique / n_samples <= MAX_PROPORTION_STRING_LABELS:
+        if (
+            n_unique / n_samples <= MAX_PROPORTION_STRING_LABELS
+            and n_unique <= MAX_NUM_STRING_LABELS
+            or n_unique <= n_bins
```

Comment:
maybe add parentheses to make it clearer as `and` and `or` are mixed

Reply:
[b4c1c46](https://github.com/huggingface/datasets-server/pull/2553/commits/b4c1c46a8aef1d692d27a29847cf4ff4268dd8ba)